File: Modulo1/Semana7/ResolucaoPraticas/PI-024/Sources/Cliente.cpp
#include "../Headers/Cliente.hpp"

int Cliente::numeroDeIdentificadores=0;

Cliente::Cliente(){

}

Cliente::Cliente(const string &nome , const string &cpf){
    Cliente::numeroDeIdentificadores++; 
    setNome(nome);
    setCpf(cpf);
    setId(Cliente::numeroDeIdentificadores);
}


void Cliente::setNumeroDeIdentificadores(const int &numero){
    Cliente::numeroDeIdentificadores = numero;
}

int Cliente::getNumeroDeIdentificadores(){
    return Cliente::numeroDeIdentificadores;
}

vector<Dependente>* Cliente::getListaDependentes(){
    return &(this->dependentes);
}

void Cliente::mostraDados(){
    cout << "ID   : " << getId() << endl;
    cout << "Nome : " << getNome() << endl;
    cout << "CPF  : " << getCpf() << endl;
}

File: java-avancado/Aula-01/CRUD-Inicial/biblioteca/src/main/java/com/biblioteca/biblioteca/model/Cliente.java
package com.biblioteca.biblioteca.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name="CLIENTE")
public class Cliente {
	
	@Id
	@GeneratedValue(strategy= GenerationType.IDENTITY)
	private Long id;
	@Column(name="NOME")
	private String nome;
    @Column(name = "CPF", unique = true) 
	private String cpf;
	
	
	public Cliente(String nome, String cpf) {
		super();
		this.nome = nome;
		this.cpf = cpf;
	}


	public Cliente() {
		super();
	}


	public Long getId() {
		return id;
	}


	public void setId(Long id) {
		this.id = id;
	}


	public String getNome() {
		return nome;
	}


	public void setNome(String nome) {
		this.nome = nome;
	}


	public String getCpf() {
		return cpf;
	}


	public void setCpf(String cpf) {
		this.cpf = cpf;
	}
	
	
	
	
	

}
